File: backend/mvb-app/src/main/java/com/service/BookingService.java
package com.service;

import com.dto.BookingRequest;
import com.model.Booking;
import java.util.List;

public interface BookingService {
	Booking bookMovie(BookingRequest request, String userEmail);

	List<Booking> getBookingsByUserEmail(String userEmail);

	List<Booking> getAllBookings(); // for admin
}

File: frontend/src/app/services/booking.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Booking } from '../models/booking.model';
import { BookingRequest } from '../models/booking-request.model';

@Injectable({ providedIn: 'root' })
export class BookingService {
  private baseUrl = 'http://localhost:9090/api/bookings';

  constructor(private http: HttpClient) { }

  bookMovie(request: BookingRequest): Observable<Booking> {
    return this.http.post<Booking>(this.baseUrl, request);
  }

  getUserBookings(): Observable<Booking[]> {
    return this.http.get<Booking[]>(`${this.baseUrl}/my`);
  }

  getAllBookings(): Observable<Booking[]> {
    return this.http.get<Booking[]>(`${this.baseUrl}/all`);
  }
}
